imap_processing/ena_maps/ena_maps.py

```diff
@@ -0,0 +1,521 @@
+            spacing_deg=self.spacing_deg,
+        )
+
+        # Unwrap the az, el, solid angle grids to series of points tiling the sky
+        az_points = self.sky_grid.az_grid.ravel()
+        el_points = self.sky_grid.el_grid.ravel()
+        self.az_el_points = np.column_stack((az_points, el_points))
+        self.solid_angle_points = self.solid_angle_grid.ravel()
+        self.num_points = self.az_el_points.shape[0]
+
+        # Initialize empty data dictionary to store map data
+        self.data_dict: dict[str, NDArray] = {}
+
+    def project_pset_values_to_map(
+        self,
+        pointing_set: PointingSet,
+        value_keys: list[str] | None = None,
+        index_match_method: IndexMatchMethod = IndexMatchMethod.PUSH,
+    ) -> None:
+        """
+        Project a pointing set's values to the map grid.
+
+        Here, the term "project" refers to the process of determining which pixels in
+        the map grid correspond to which pixels in the pointing set grid, and then
+        binning the values at those indices from the pointing set to the map.
+
+        Parameters
+        ----------
+        pointing_set : PointingSet
+            The pointing set containing the values to project to the map.
+        value_keys : list[tuple[str, IndexMatchMethod]] | None
+            The keys of the values to project to the map.
+            Ex.: ["counts", "flux"]
+            data_vars named each key must be present, and of the same dimensionality in
+            each pointing set which is to be projected to the map.
+            Default is None, in which case all data_vars in the pointing set are used.
+        index_match_method : IndexMatchMethod, optional
+            The method of index matching to use for all values.
+            Default is IndexMatchMethod.PUSH.
+
+        Raises
+        ------
+        ValueError
+            If a value key is not found in the pointing set.
+        """
+        if value_keys is None:
+            value_keys = list(pointing_set.data.data_vars.keys())
+
+        for value_key in value_keys:
+            if value_key not in pointing_set.data.data_vars:
+                raise ValueError(f"Value key {value_key} not found in pointing set.")
+
+        # Determine the indices of the sky map grid that correspond to
+        # each pixel in the pointing set.
+        if index_match_method is IndexMatchMethod.PUSH:
+            matched_indices_push = match_coords_to_indices(
+                input_object=pointing_set,
+                output_object=self,
+            )
+
+        for value_key in value_keys:
+            # If multiple spatial axes present
+            # (i.e (az, el) for rectangular coordinate PSET),
+            # flatten them in the values array to match the raveled indices
+            raveled_pset_data = np.reshape(
+                np.array(pointing_set.data[value_key]),
+                (pointing_set.num_points, -1),
+            )
```

Comment:
Can you reshape the data array directly to avoid creating copies here?
```suggestion
            raveled_pset_data = pointing_set.data[value_key].reshape(pointing_set.num_points, -1)
```

Reply:
Sort of - There's no `xr.DataArray.reshape()` method, but I can do a `pointing_set.data[value_key]`**.data**`.reshape(pointing_set.num_points, -1)`. I'm not actually sure if that performs a copy. I think that it doesn't, since `.data` should just access the internal numpy representation.